board/board.h

```diff
@@ -0,0 +1,87 @@
+// ///////////////////////////////////////////////////////////// //
+// Hardware abstraction layer for all different supported boards //
+// ///////////////////////////////////////////////////////////// //
+#include "boards/common.h"
+int usb_power_mode = USB_POWER_NONE;
+
+// ///// Function definitions ///// //
+typedef void (*board_init)(void);
+typedef void (*board_enable_can_transciever)(uint8_t transciever, bool enabled);
+typedef void (*board_enable_can_transcievers)(bool enabled);
+typedef void (*board_set_led)(uint8_t color, bool enabled);
+typedef void (*board_set_usb_power_mode)(uint8_t mode);
+typedef void (*board_set_esp_gps_mode)(uint8_t mode);
+typedef void (*board_set_can_mode)(uint8_t mode);
+
+typedef struct {
+  const char *board_type;
+  const uint8_t num_transcievers;
+  board_init init;
+  board_enable_can_transciever enable_can_transciever;
+  board_enable_can_transcievers enable_can_transcievers;
+  board_set_led set_led;
+  board_set_usb_power_mode set_usb_power_mode;
+  board_set_esp_gps_mode set_esp_gps_mode;
+  board_set_can_mode set_can_mode;
+} board;
+
+// ///// Board definition ///// //
+#ifdef PANDA
+  #include "boards/white.h"
+  #include "boards/grey.h"
+  #include "boards/black.h"
+#else
+  #include "boards/pedal.h"
+#endif
+
+// ///// Board detection ///// //
```

Comment:
@robbederks , no need to make  GREY val 1. This is fine, I'll change boardd. 
Also, there should be a comment here to say that this enum shall match the one in cereal/log.capnp.

Since the pedal is not a considered a panda, I recommend the following names:
```
#define HW_TYPE_UNSUPPORTED 0
#define HW_TYPE_WHITE_PANDA 1
#define HW_TYPE_GREY_PANDA 2
#define HW_TYPE_BLACK_PANDA 3
#define HW_TYPE_PEDAL 4
```

Reply:
Ok, great, I'll change it right away!